aws_lambda_powertools/utilities/feature_toggles/configuration_store.py

```diff
@@ -0,0 +1,207 @@
+# pylint: disable=no-name-in-module,line-too-long
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+from botocore.config import Config
+
+from aws_lambda_powertools.utilities.parameters import AppConfigProvider, GetParameterError, TransformParameterError
+
+TRANSFORM_TYPE = "json"
+FEATURES_KEY = "features"
+RULES_KEY = "rules"
+DEFAULT_VAL_KEY = "default_value"
+RESTRICTIONS_KEY = "restrictions"
+RULE_NAME_KEY = "name"
+RULE_DEFAULT_VALUE = "default_value"
+RESTRICTION_KEY = "key"
+RESTRICTION_VALUE = "value"
+RESTRICTION_ACTION = "action"
+
+
+class ACTION(str, Enum):
+    EQUALS = "EQUALS"
+    STARTSWITH = "STARTSWITH"
+    ENDSWITH = "ENDSWITH"
+    CONTAINS = "CONTAINS"
+
+
+class ConfigurationException(Exception):
+    pass
+
+
+class ConfigurationStore:
+    def __init__(
+        self, environment: str, service: str, conf_name: str, cache_seconds: int, region_name: Optional[str] = None
+    ):
+        """constructor
+
+        Args:
+            environment (str): what appconfig environment to use 'dev/test' etc.
+            service (str): what service name to use from the supplied environment
+            conf_name (str): what configuration to take from the environment & service combination
+            cache_seconds (int): cache expiration time, how often to call AppConfig to fetch latest configuration
+            region_name (str): aws region where the configuration resides in
+        """
+        self._cache_seconds = cache_seconds
+        self._conf_name = conf_name
+        config = None
+        if region_name is not None:
+            config = Config(region_name=region_name)
+        self._conf_store = AppConfigProvider(environment=environment, application=service, config=config)
+
+    def _validate_json_schema(self, schema: str) -> bool:
+        ## todo
+        return True
+
+    def _match_by_action(self, action: str, restriction_value: Any, context_value: Any) -> bool:
+        mapping_by_action = {
+            ACTION.EQUALS.value: lambda a, b: a == b,
+            ACTION.STARTSWITH.value: lambda a, b: a.startswith(b),
+            ACTION.ENDSWITH.value: lambda a, b: a.endswith(b),
+            ACTION.CONTAINS.value: lambda a, b: a in b,
+        }
+
+        try:
+            func = mapping_by_action.get(action, lambda a, b: False)
+            return func(context_value, restriction_value)
+        except Exception:
+            return False
+
+    def _handle_rules(
```

Comment:
Would a memoize pattern ( https://docs.python.org/3/library/functools.html - cache ) be useful to avoid runtime executions with same context/rules? The benefit would only be seen from executions on the same lambda invocation, but anyway, can be useful for example in batch operations.

Reply:
yeah I'm familiar with it. However, depending on your schema size, the rules list can be very large. I'm not sure we'd want to use it for cache at that point. what do you think, is that an issue?